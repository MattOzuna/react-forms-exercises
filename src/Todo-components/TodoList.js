import { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

const TodoList = () => {
  const [Todos, setTodos] = useState([]);

  const addTask = (task) => {
    setTodos((Todos) => [...Todos, task]);
  };

  const deleteTask = (idx) => {
    setTodos(Todos.toSpliced(idx, 1));
  };

  return (
    <div className="TodoList">
      <h1>Todo List</h1>
      {Todos.map((task, idx) => (
        <Todo task={task} deleteTask={deleteTask} id={idx} key={idx} />
      ))}
      <NewTodoForm addTask={addTask} />
    </div>
  );
};

export default TodoList;
