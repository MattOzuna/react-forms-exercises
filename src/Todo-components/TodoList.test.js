import { render, asFragment } from "@testing-library/react"
import TodoList from "./TodoList"
import '@testing-library/jest-dom';


it("should render", ()=>{
    render(<TodoList />);
})

it('should match snapshot', () =>{
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
})