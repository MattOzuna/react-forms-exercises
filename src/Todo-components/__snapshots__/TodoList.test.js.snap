// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h1>
      Todo List
    </h1>
    <form
      class="NewTodoForm"
    >
      <label
        for="task"
      >
        New Task: 
      </label>
      <input
        id="task"
        name="task"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
